src/imap/mod.rs

```diff
@@ -675,25 +688,38 @@ impl Imap {
             {
                 // Trigger download and processing for this message.
                 uids.push(current_uid);
-            } else if read_errors == 0 {
-                // No errors so far, but this was skipped, so mark as last_seen_uid
-                new_last_seen_uid = Some(current_uid);
             }
         }
 
         // check passed, go fetch the emails
-        let (new_last_seen_uid_processed, error_cnt) = self
+        let (_, error_cnt) = self
             .fetch_many_msgs(context, &folder, uids, fetch_existing_msgs)
             .await;
         read_errors += error_cnt;
 
-        // determine which last_seen_uid to use to update  to
-        let new_last_seen_uid_processed = new_last_seen_uid_processed.unwrap_or_default();
-        let new_last_seen_uid = new_last_seen_uid.unwrap_or_default();
-        let last_one = new_last_seen_uid.max(new_last_seen_uid_processed);
+        // determine which uid_next to use to update  to
+        let new_uid_next = max(
+            largest_uid.unwrap_or_default() + 1,
+            (self.config.selected_mailbox.as_ref())
+                .and_then(|m| m.uid_next)
+                .unwrap_or(1),
+        );
 
-        if last_one > last_seen_uid {
-            set_config_last_seen_uid(context, &folder, uid_validity, last_one).await;
+        // TODO:
+        // The error_cnt == 0 condition is here because dc_receive_imf() returns an `Err` value only on recoverable
+        // errors, otherwise it just logs an error. This means that `fectch_many_msgs` will only increase the `read_errors`
+        // variable on recoverable errors and we can check the `error_cnt` to see if there was an error
+        // and if there was, not update uid_next so that the message is fetched again next time.
+        //
+        // Problem: There might be a message
+        // that for whatever reason causes an `Err` value to be returned every time. Before my PR, the solution was:
+        // Update the uid_next to the largest message uid that had NOT failed parsing. Not perfect either but maybe I should just
```

Comment:
Old logic looks ok, except that `max` in `new_last_seen_uid.max(new_last_seen_uid_processed);` should be `min`? Updating to the last one processed looks good to me.

Reply:
> except that max in new_last_seen_uid.max(new_last_seen_uid_processed); should be min?

It's been correct, `x.max(y)` [does the exact opposite of what one thinks it does on first look](https://internals.rust-lang.org/t/am-i-the-only-one-confused-by-a-min-b-and-a-max-b/13252). Probably best to avoid it completely and use `max(x,y)`

Reply:
I thought it should be `min(new_last_seen_uid, new_last_seen_uid_processed)`. Actually `max` is right here.

Anyway, I think implementing old logic (pushing last seen as far the last successfully processed message) is the right thing to do.

Reply:
I implemented the old logic again and tried to improve code readability